### DISCUSSION TITLE
Where does sound emulation stand for Microprose's 1942?

### DISCUSSION THREAD

**Discussion by PoulpSquad** [dosbox-staging/dosbox-staging#3768] [General]
Hello @ all,

I was wondering if there is any improvement towards the sound emulation in the title "1942 Gold" from Microprose.

Sound on that game has been broken for years, and I am wondering if maybe there is a way to fix it in recent builds?

The only reliable way to have working sound is to disable XMS/EMS and load JEMMEX.EXE. Is this still true, or am I missing something?

Thank you for your time reading this.

**Comment by johnnovak** [comment]
Well, you tell us @PoulpSquad 😄 Please test it with the latest dev build, and if you believe there is a problem, please raise a defect. But if there's a workable solution already, what's wrong with that?

**Comment by PoulpSquad** [comment]
Nothing, besides the fact it's ugly ^^

The installer runs fine in Windows 3.x, but tries to switch to DOS to run the game proper, and that fails.

Running INSTALL.EXE from the CD trips on a "cannot open file" error.

Copying files from CD to hard drive works, but sound is horribly mangled and non-functional without the help of an external EMM, and we don't get to access the gold content.

I'll toy with it a little longer before opening a proper case, although if running JEMMEX.EXE is deemed a good enough solution, I'll abstain from cluttering.

**Reply by johnnovak** [reply]
> Nothing, besides the fact it's ugly ^^

Well, we're after practicality here. If the game works, it works—"ugly" or "beautiful" doesn't enter the picture, as I see it 😎 

Remember CONFIG.SYS hackery on real hardware to get games working _at all_? *That* was ugly 😆

**Reply by PoulpSquad** [reply]
> Remember CONFIG.SYS hackery on real hardware to get games working at all? That was ugly

All too well xD

And yes, you're right. I'll just put that one on the weird pile of DOS game.

**Reply by Torinde** [reply]
I agree it shouldn't be priority since there is _a_ solution. But back in the day JEMMEX didn't exist, so there is also some emulation gap here, e.g. the game should be able to install/run without modern tools?

**Reply by PoulpSquad** [reply]
On the other hand, stuff like DOSBox didn't exist, either.

AFAIR 1942 used to work just fine on my 486DX from 1992 with regular DOS 5.0 and brave good old EMM386.

DOSBox has solved many (MANY!) issues, and has added some others... I'm not a fan of this hacky solution and I think there's something amiss with EMS in DOSBox, but since I'm next to illiterate, messing with Jemm will have to do.

Earlier (beta) versions of Jemm didn't remove themselves from memory, so a DOSBox restart was needed after each session.

**Comment by PoulpSquad** [comment]
In case anybody runs into trouble with this game, here's a procedure that worked for me:

- Install game on Windows. Sound selection doesn't seem to have any importance.
- Copy INSTALL.EXE and INSTALL.INF from the CD to the installation directory (C:\MPS\PAWGOLD by default).
- Exit Windows and run INSTALL.EXE and chose the appropriate sound hardware.
- Download Jemm 5.84 from [https://github.com/Baron-von-Riedesel/Jemm](https://github.com/Baron-von-Riedesel/Jemm).
- Unpack the archive's content to a directory.
- Make sure that xms=false, ems=false and umb=false.
- Install Jemm: jemmex.exe d=128 frame=e000 noinvlpg noram load.
- Run PAW with 1942FULL.EXE.
- Remove JemmEx from memory with jemmex.exe unload.

If UMBs are enabled by removing "noram" and a TSR is loaded there, Jemm won't be able to remove itself from memory and Staging will have to be restarted to run other games that don't tolerate an EMM (Ultima VII, Sperpent Isle, Comanche etc.).

The "noinvlpg" parameter is apparently required when Jemm is running inside emulators (QEMU, 86Box, PCEm etc.).

The Windows side of PAW is semi-working. I wasn't able to run the game from Windows, but the ships/planes database is functional.

Props to Baron-von-Riedesel for Jemm, by the way!

**Reply by johnnovak** [reply]
Thanks @PoulpSquad, super informative, we'll add this to the wiki.

**Comment by PoulpSquad** [comment]
Great, thank you johnnovak.

That's the last one in my modest collection of DOS games I used to play in my youth that needed fixing.

They are now all working great in Staging, without having to leave the emulator/use special configs/use a launcher.

Kids nowadays have no idea how good and easy they have it!

**Comment by johnnovak** [comment]
The game has the sound working with JEMM now as per our Game Issues wiki, so closing.
